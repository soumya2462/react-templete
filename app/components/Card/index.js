/**
 *
 * Card
 *
 */

import { Card as AntCard } from 'antd';
import React from 'react';
import styled from 'styled-components';

const CustomCard = styled(AntCard)`
  && {
    margin: 20px 0;
    max-width: ${(props) => props.maxwidth};
    color: ${(props) => props.color};
    ${(props) => props.color && `color: ${props.color}`};
  }
`;

const Card = (props) => <CustomCard data-testid="card" {...props} />;

Card.propTypes = {};

export default Card;
