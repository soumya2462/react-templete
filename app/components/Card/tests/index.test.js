/**
 *
 * Tests for Card
 *
 */

import React from 'react';
// import { fireEvent } from '@testing-library/dom'
import { renderWithIntl } from '@utils/testUtils';
import Card from '../index';

describe('<Card />', () => {
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<Card />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should contain 1 Card component', () => {
    const { getAllByTestId } = renderWithIntl(<Card />);
    expect(getAllByTestId('card').length).toBe(1);
  });
});
