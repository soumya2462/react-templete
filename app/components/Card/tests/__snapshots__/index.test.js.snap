// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="ant-card ant-card-bordered Card__CustomCard-sc-7y2i64-0 hWoWqx"
      data-testid="card"
    >
      <div
        class="ant-card-body"
      />
    </div>
  </div>
</body>
`;
